src/Shared/Components/MaterialHistory/MaterialHistory.component.tsx

```diff
@@ -32,34 +35,35 @@ const MaterialHistory = ({
         }
     }
 
-    const getMaterialHistoryMapWithTime = () => {
-        const historyTimeMap = {}
-
-        material.history?.forEach((history) => {
-            const newDate = history.date.substring(0, 16)
-
-            if (!historyTimeMap[newDate]) {
-                historyTimeMap[newDate] = []
-            }
-            historyTimeMap[newDate].push(history)
-        })
+    const materialHistoryMapWithTime = useMemo(
+        () =>
+            material.history.reduce<Record<string, MaterialHistoryType[]>>((acc, historyElem: MaterialHistoryType) => {
+                const isWebhook = material.type === SourceTypeMap.WEBHOOK
+                const newDate = isWebhook
+                    ? getWebhookDate(material.type, historyElem).substring(0, 16)
+                    : historyElem.date.substring(0, 16)
+                if (!acc[newDate]) {
+                    acc[newDate] = []
+                }
+                acc[newDate].push(historyElem)
+                return acc
+            }, {}),
+        [material.history, material.type],
+    )
 
-        return historyTimeMap
-    }
     // Retrieve the history map
-    const materialHistoryMapWithTime = getMaterialHistoryMapWithTime()
     // Retrieve the keys of the history map
     const dateKeys = Object.keys(materialHistoryMapWithTime)
 
     return (
         // added for consistent typing
         // eslint-disable-next-line react/jsx-no-useless-fragment
-        <>
+        <div className="h-100 flex-grow-1 dc__overflow-scroll">
             {dateKeys.map((date) => {
                 const historyList = materialHistoryMapWithTime[date]
                 return (
-                    <>
-                        {!isCommitInfoModal && material.type !== SourceTypeMap.WEBHOOK && (
+                    <div key={date} className="flexbox-col dc__gap-12 py-12 px-16">
```

Comment:
Since we are using substring on the date, there might be a possibility of duplicate keys here. Pls check once